source/write/gridfs.txt

```diff
@@ -0,0 +1,309 @@
+and the number of documents to skip before returning. To view a list of available
+options, see the `API documentation <{+api+}/method/MongoDBGridFSBucket-find/#parameters>`__.
+
+.. _gridfs-download-files:
+
+Download Files
+--------------
+
+You can download files from your MongoDB database by using the
+``MongoDB\GridFS\Bucket::openDownloadStreamByName()`` method to
+create a download stream.
+
+This example uses a download stream to perform the following actions:
+
+- Selects a GridFS file named ``'my_file'``, uploaded in the
+  :ref:`gridfs-upload-files` example, and opens it as a readable stream 
+- Calls the ``stream_get_contents()`` method to read the contents of ``'my_file'``
+- Prints the file contents
+- Calls the ``fclose()`` method to close the download stream pointing to ``'my_file'`` 
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/write/gridfs.php
+      :start-after: start download files name
+      :end-before: end download files name
+      :language: php
+      :dedent:
+
+   .. output::
+      :visible: false
+
+      "Data to store"
+
+.. note::
+
+   If there are multiple documents with the same file name,
+   GridFS will stream the most recent file with the given name (as
+   determined by the ``uploadDate`` field).
+
+Alternatively, you can use the ``MongoDB\GridFS\Bucket::openDownloadStream()``
+method, which takes the ``_id`` field of a file as a parameter:
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start download files id
+   :end-before: end download files id
+
+.. note::
+
+   The GridFS streaming API cannot load partial chunks. When a download
+   stream needs to pull a chunk from MongoDB, it pulls the entire chunk
+   into memory. The 255-kilobyte default chunk size is usually
+   sufficient, but you can reduce the chunk size to reduce memory
+   overhead.
+
+.. _gridfs-rename-files:
+
+Rename Files
+------------
+
+Use the ``MongoDB\GridFS\Bucket::rename()`` method to update the name of
+a GridFS file in your bucket. You must specify the file to rename by its
+``_id`` field rather than its file name.
+
+The following example shows how to update the ``filename`` field to
+``'new_file_name'`` by referencing a document's ``_id`` field:
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start rename files
+   :end-before: end rename files
+
+.. note::
+
+   The ``rename()`` method supports updating the name of only one file at
+   a time. To rename multiple files, retrieve a list of files matching the
+   file name from the bucket, extract the ``_id`` field from the files you
+   want to rename, and pass each value in separate calls to the ``rename()``
+   method.
+
+.. _gridfs-delete-files:
+
+Delete Files
+------------
+
+Use the ``MongoDB\GridFS\Bucket::delete()`` method to remove a file's collection
+document and associated chunks from your bucket. This effectively deletes the file.
```

Comment:
May be worth clarifying that this is going to delete a _revision_ of a file. If another `fs.files` document exists for that filename, it can still be read. And if users aren't aware of GridFS' revision behavior and selecting the most recent by default, they could be surprised to discover that they can still query for a filename after calling `delete()`. The clarification should help avoid that.

Note that this also applies to `rename()`. Common advice for both may be telling users to collect all IDs for a filename if they intend to rename or delete all of its revisions. I don't think you need example code for that (can be left as an exercise for the reader).

Reply:
I modified the note below the code examples with this information

Reply:
I see the new "File Revisions" notes for renames/deletes. Is there a DOCSP ticket to track future work of documenting the `revision` option for download methods? That seems like it'd be generally relevant to all driver GridFS tutorials.

Reply:
Not currently, although a few pages have a section on revisions (like [Java](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/gridfs/#file-revisions)) or mention it somewhere on the page